publish/src/command-utils/owner-actions.js

```
@@ -0,0 +1,142 @@
+'use strict';
+
+const ethers = require('ethers');
+const { gray, yellow } = require('chalk');
+
+const { logTx } = require('./ui-utils');
+
+const {
+	getSafeInstance,
+	getSafeNonce,
+	getSafeTransactions,
+	checkExistingPendingTx,
+	getNewTransactionHash,
+	saveTransactionToApi,
+	getSafeSignature,
+} = require('./safe-utils');
+
+const KIND = { safe: 'SAFE', eoa: 'EAO' };
+
+const getSignerData = async ({ signerKind, providerUrl, newOwner }) => {
+	const signerData = {};
+	if (signerKind === KIND.safe) {
+		// new owner should be gnosis safe proxy address
+		signerData.protocolDaoContract = getSafeInstance(providerUrl, newOwner);
+
+		// get protocolDAO nonce
+		signerData.currentSafeNonce = await getSafeNonce(signerData.protocolDaoContract);
+
+		if (!signerData.currentSafeNonce) {
+			console.log(gray('Cannot access safe. Exiting.'));
+			process.exit();
+		}
+
+		console.log(
+			yellow(`Using Protocol DAO Safe contract at ${signerData.protocolDaoContract.address}`)
+		);
+	}
+
+	return signerData;
+};
+
+const getStagedTransactions = async ({ signerKind, signerData, network }) => {
+	let stagedTransactions;
+	if (signerKind === KIND.safe) {
+		stagedTransactions = await getSafeTransactions({
+			network,
+			safeAddress: signerData.protocolDaoContract.address,
+		});
+	}
+
+	return stagedTransactions;
+};
+
+const txAlreadyExists = async ({
+	signerKind,
+	signerData,
+	stagedTransactions,
+	target,
+	encodedData,
+}) => {
+	if (signerKind === KIND.safe) {
+		return checkExistingPendingTx({
+			stagedTransactions,
+			target,
+			encodedData,
+			currentSafeNonce: signerData.currentSafeNonce,
+		});
+	}
```

Same here, feels like other cases are not handled and might fail silently in the future.

---

dito